### ISSUE TITLE
ENGIN-X: Migrate to IBEX

### ISSUE THREAD

**Issue by kjwoodsISIS** [ISISComputingGroup/IBEX#2399]
As an instrument scientist, I would like to migrate ENGIN-X to use the IBEX instrument control system.

#### Acceptance Criteria
1. IBEX is installed on ENGIN-X
1. Configuration files matching the current SECI config files are created
1. IBEX is configured for the devices on ENGIN-X.
1. ENGIN-X scripts are converted to genie_python
1. ENGIN-X scientists are trained in how to use IBEX
1. IBEX is tested on ENGIN-X

#### Notes
1. The devices on ENGIN-X are listed on the [ENGIN-X Instrument Details](https://github.com/ISISComputingGroup/IBEX/wiki/ENGIN-X-Instrument-Details) list
1. The following tickets should be implemented prior to deploying IBEX on ENGIN-X
   - [x] #2111
   - [x] #2112
   - [x] #2113
   - [x] #2114
   - [x] #2115
   - [x] #2235
   - [x] #2338
   - [x] #2378
   - [x] #2400
   - [x] #2401
   - [x] #2462
   - [ ] #2470
   - [x] #2489
   - [x] #2520
   - [x] #2521
   - [x] #2522
1. The following tickets are "nice-to-have", but are not pre-requistites for deploying IBEX on ENGIN-X
   - [ ] #2116
1. Use the [Migrate-an-Instrument-Control-PC](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Migrate-an-Instrument-Control-PC) page to guide you through the migration process. The following activities need to be completed as part of the migration process:
   - [x] Ask Facilities IT (Anthony Shuttle) to create a private network for ENGIN-X.  This is required so that we can use TCP/IP to communicate with the Galils. Facilities IT will need notice of about 4-6 weeks.
      - 27/07/2017: Confirmed by Anthony; scheduled to happen mid-August.
   - [x] Check that NDXENGINX has sufficient disk space to install IBEX
      - C: drive - Total: 127.00 GB, Free: 92.67 GB on 28/07/2017.
      - Fresh install of IBEX requires 16.0GB + 20GB growth room.  92.7GB free is sufficient space.
   - [x] Check whether ENGIN-X uses 5- or 8- digit run-numbers
      - [x] ENGIN-X uses 8-digit run numbers
   - [ ] Create a virtual instrument for testing purposes
      - [ ] Set up a test system, say, NDXENGINXTEST.
      - [x] create IBEX configurations to correspond to their SECI equivalents
      - [x] create genie-python scripts to correspond to their Open-Genie equivalents
      - [ ] test these configurations & scripts
      - [ ] check that all the devices previously supported by SECI (on ENGIN-X) are supported by IBEX
   - [x] arrange one or more training sessions with the scientists to help them understand how to use IBEX
      - [ ] demonstrate the configurations & scripts (created above) to the instrument scientists and check that they are happy with the migrated configurations & scripts
   - [x] Check location of ENGIN-X wiring tables and update if necessary (see comment from Gareth below)

**Comment by kjwoodsISIS** [comment]
> From: Howells, Gareth (STFC,RAL,ISIS)
> Sent: 21 July 2017 16:20
> To: Woods, Kevin (Tessella,RAL,ISIS)
> Subject: ENGINX DAE wiring tables
> 
> Dear Kevin,
>  
> Before I forget, just a quick email to let you know that I think that ENGINX DAE wiring tables might be across the local machine and on a network drive. The locations might be :
>  
> 1)      Local: \\ndxenginx\c$\LabVIEW Modules\dae\Instrument Tables
> 2)      Network : \\isis\inst$\NDXENGINX\Instrument\templates
>  
> Not sure which wiring tables are used, but I seem to recall in the past that the LabVIEW DAE vi looked on a network location. It might not even be the one listed above. The instrument scientists will probably know the details of what is current.
>  
> An example of what I am referring to is shown below from the file : \\ndxenginx\c$\Data\Export only\logs\icp\log\icp-2017-05-31-Wed.log
>  
> 2017-05-31 10:39:23  Wiring table D:\templates\enginx05_w.dat is unchanged - reload skipped
> 2017-05-31 10:39:23  Detector table "D:\templates\enginx02_transdet_d-tab2.dat" is unchanged - reload skipped
> 2017-05-31 10:39:23  Spectra table D:\templates\enginx02_transdet_s.dat is unchanged - reload skipped
>  
> It might not be important, but just in case ….
> 
> Regards,
>  
> Gareth

**Comment by Tom-Willemsen** [comment]
Would be better to do after #2462

**Comment by DominicOram** [comment]
After talking to the instrument scientists the configs mostly use:
* StressRig+FurnaceAndCryo (will use this even if they are not using their furnace)
* Cybaman+Cryo
* As a core they will always have positioner, jaws and choppers

**Comment by Tom-Willemsen** [comment]
The only things remaining on this ticket are:
- [ ] Finish off #2470 (in progress, mostly complete)
- [ ] Demonstrate the configurations and scripts to scientists
- [ ] Test the configurations and scripts

I think this ticket can be reviewed as it stands, pending final testing and acceptance from scientists.

**Comment by John-Holt-Tessella** [comment]
- [ ]  Check names on JAWS set names for JAW 2 East and West seem to be called JAW 3 and JAW 4
